stl/inc/xmemory

```diff
@@ -2621,6 +2621,12 @@ _NODISCARD _Elem* _UIntegral_to_buff(_Elem* _RNext, _UTy _UVal) { // used by bot
 }
 _STD_END
 
+#if _HAS_CXX17

```

Comment:
Maybe add a comment like:
```c++
// Don't deduce `emplace_back` return type to guard against omniconvertible types
// and prevent ODR violation with C++14//C++17 mixing; see GH-4963
```

Reply:
We talked about this at the weekly maintainer meeting and we believe that the usage and intent of this macro is sufficiently clear that it doesn't need to be commented. Our git history and test coverage should be sufficient to remember why we did this.

Importantly, this area isn't being actively modified, and future containers and container-like entities will not have varying return types, so there won't be increasing usage of this macro over time.